pkg/sql/logictest/testdata/logic_test/serializable_eager_restart

```diff
@@ -29,13 +32,17 @@ SELECT * FROM t
 user root
 
 # The insert increases the candidate timestamp, but a restart won't occur
-# until an EndTransaction.
+# until RELEASE SAVEPOINT.
 statement ok
 INSERT INTO t(a) VALUES (1)
 
-# Commit will send an EndTransaction which will result in a retry error.
+# RELEASE SAVEPOINT.
 statement error pgcode 40001 retry txn.*
-COMMIT
+RELEASE SAVEPOINT cockroach_restart
```

Comment:
I found in `TestLogic/default/serializable_eager_restart` and `TestLogic/distsql/serializable_eager_restart`, it reports error like `expected "retry txn.*", but found <nil>`.

Actually, it seems only in parallel condition, it will return an retriable error.

Reply:
That is unexpected, right? Can you debug? You can run just a single logic test with something like (change the test names):
```
make testshort PKG=./pkg/sql/logictest  TESTTIMEOUT=2m TESTFLAGS="-v -show-logs -show-sql --vmodule=executor=2" TESTS=TestLogic/distsql-disk/postgresjoin
```

If you fix the server code, I can look too.